CHANGES.rst

```diff
@@ -7,6 +7,8 @@ New Features
 Other Changes and Additions
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+- Remove astropy_helpers from the package infrastructure [#717]
```

Comment:
It might make sense to mention that this also changes how the tests are run and the docs are built.